modules/mono/glue/GodotSharp/GodotSharp/Core/Projection.cs

```diff
@@ -0,0 +1,822 @@
+            return new Vector4(
+                proj.x.x * v.x + proj.y.x * v.y + proj.z.x * v.z + proj.w.x * v.w,
+                proj.x.y * v.x + proj.y.y * v.y + proj.z.y * v.z + proj.w.y * v.w,
+                proj.x.z * v.x + proj.y.z * v.y + proj.z.z * v.z + proj.w.z * v.w,
+                proj.x.w * v.x + proj.y.w * v.y + proj.z.w * v.z + proj.w.w * v.w
+            );
+        }
+
+        /// <summary>
+        /// Returns <see langword="true"/> if the projections are exactly equal.
+        /// </summary>
+        /// <param name="left">The left projection.</param>
+        /// <param name="right">The right projection.</param>
+        /// <returns>Whether or not the projections are exactly equal.</returns>
+        public static bool operator ==(Projection left, Projection right)
+        {
+            return (left.x == right.x) && (left.y == right.y) && (left.z == right.z) && (left.w == right.w);
+        }
+
+        /// <summary>
+        /// Returns <see langword="true"/> if the projections are not exactly equal.
+        /// </summary>
+        /// <param name="left">The left projection.</param>
+        /// <param name="right">The right projection.</param>
+        /// <returns>Whether or not the projections are not exactly equal.</returns>
+        public static bool operator !=(Projection left, Projection right)
+        {
+            return !(left == right);
+        }
+
+        /// <summary>
+        /// Access whole columns in the form of <see cref="Vector4"/>.
+        /// </summary>
+        /// <param name="column">Which column vector.</param>
+        public Vector4 this[int column]
+        {
+            get
+            {
+                switch (column)
+                {
+                    case 0:
+                        return x;
+                    case 1:
+                        return y;
+                    case 2:
+                        return z;
+                    case 3:
+                        return w;
+                    default:
+                        throw new IndexOutOfRangeException();
+                }
+            }
+            set
+            {
+                switch (column)
+                {
+                    case 0:
+                        x = value;
+                        return;
+                    case 1:
+                        y = value;
+                        return;
+                    case 2:
+                        z = value;
+                        return;
+                    case 3:
+                        w = value;
+                        return;
+                    default:
+                        throw new IndexOutOfRangeException();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Access single values.
+        /// </summary>
+        /// <param name="column">Which column vector.</param>
+        /// <param name="row">Which row of the column.</param>
+        internal real_t this[int column, int row]
```

Comment:
This operator is public in `Basis`, we may want to consider doing the same here but it shouldn't block this PR.